source/configuration_manual/ssl_configuration.rst

```diff
@@ -48,3 +48,15 @@ If you need different SSL certificates for IMAP and POP3 protocols, you can put
 Dovecot supports also using TLS SNI extension for giving different SSL certificates based on the server name when using only a single IP address, but the TLS SNI isn't yet supported by all clients so that may not be very useful. 
 
 It's anyway possible to configure it by using ``local_name imap.example.com {}`` blocks.
+
+JA3 hash
+--------
+
+.. versionadded:: v2.4.0;v3.0.0
+
+Dovecot supports calculating `JA3 hash <https://engineering.salesforce.com/tls-fingerprinting-with-ja3-and-ja3s-247362855967/>`__ for checking client TLS implementation.
+This adds ``ssl_ja3`` and ``ssl_ja3_hash`` to :ref:`login variables <variables-login>`, to be used with :dovecot_core:ref:`login_log_format_elements`
+and ``ssl_ja3_hash`` for :ref:`authentication variables <variables-auth>`, to be used with e.g. :ref:`authentication-auth_policy`.
+
+Please note that since IMAP does not usually support ALPN, hash databases used with http protocol are generally not suitable sources for hashes. This is usually because ALPN and NPN
```

Comment:
@cmouse ping

Reply:
umm it is the case. http hashes have alpn and npn extensions and imap ones do not.

Reply:
Could be written somehow more clearly for people who don't already understand this. To me it was very confusing at least. How about:

Please note that JA3 databases used with HTTP generally include ALPN and NPN values. This makes the databases unsuitable for IMAP, because IMAP clients usually don't support them.

"extensions 16 and 13172" - is this important? Are they just numbers for the ALPN/NPN?

Reply:
yes, but they also are the exact numbers you need to remove from the HTTP JA3 value to make it IMAP compatible. So the JA3 string in HTTP would say `769,47–53–5–10–49161–49162–49171–49172–50–56–19–4,0–10–11-16-13172,23–24–25,0` (for example...) and you could remove the extensions from there and hash it to make it imap compatible, most likely.

Reply:
maybe it's better now?

Reply:
Wouldn't hurt to add this specific example before and after the 16-13172 removal.